src/pk_core.c

```diff
@@ -1217,6 +1217,15 @@ Var varBitRshift(PKVM* vm, Var v1, Var v2) {
   return VAR_NULL;
 }
 
+Var varBitNot(PKVM* vm, Var v) {
+  int64_t i;
+
+  if (isInteger(v, &i))
+    if (!validateInteger(vm, v, &i, "Unary operand")) return VAR_NULL;
+
+  return VAR_NUM((double)(~i));
```

Comment:
`isInteger` returns true if `v` is an integer, if it's not, the `i` will be uninitialized ( and this will not set an error). After that it will return the noise value (cause UB) at the line 1226.

Inside the `isInteger` scope, we already know that `v` an integer, there is no point validating it. and if it's an integer it'll always return `VAR_NULL` (line 1224).

We need `VAR_NUM((double)(~i))` to be returned when the `v` is an integer otherwise it should set a runtime error, (the `validateInteger` will set the error if `v` isn't an integer) and return VAR_NULL.

If you're still troubling to understand, feel free to ask 😄.

Reply:
Oh, It'll always reach line 1226, regardless of it's type. could you change this to.
```c
if (!validateInteger(vm, v2, &i, "Unary operand")) return VAR_NULL;
return VAR_NUM((double)(~i));
```